### PR TITLE
feat: Add HowTo re new K8s clusters with Magnum

### PR DESCRIPTION
**Pull request by colder-is-better**
We show, step by step, how to create new Kubernetes clusters with OpenStack Magnum, using either the Cleura Cloud Management Panel or the OpenStack CLI.

### CODE REVIEW

File: docs/howto/openstack/magnum/new-k8s-cluster.md [hunk 1]
```diff
@@ -0,0 +1,308 @@
...
+the OpenStack CLI. Let us demonstrate the creation of a Kubernetes 
+cluster following both approaches.
+
+## Prerequisites
+
+First and foremost, you need an [account in {{brand}} 
+Cloud](/howto/getting-started/create-account). If you prefer to work 
+with the OpenStack CLI, go ahead and [enable it 
+first](/howto/getting-started/enable-openstack-cli). Then, make sure 
+you have the appropriate Magnum client Python module installed. Use
```
Comment (fghaas) [inline docs/howto/openstack/magnum/new-k8s-cluster.md:15]: I would use:

```markdown
Then, in addition to the Python `openstackclient` module, make sure you also install
the corresponding plugin module for Magnum.
```

File: docs/howto/openstack/magnum/new-k8s-cluster.md [hunk 2]
```diff
@@ -0,0 +1,308 @@
...
+first](/howto/getting-started/enable-openstack-cli). Then, make sure 
+you have the appropriate Magnum client Python module installed. Use 
+either the package manager of your operating system or `pip`:
+
+=== "Debian/Ubuntu"
+	```bash
+	apt install python3-magnumclient
+	```
+=== "Mac OS X with Homebrew"
+	This Python module is unavailable via `brew`, but you can have
```
Comment (fghaas) [inline docs/howto/openstack/magnum/new-k8s-cluster.md:23]: I'd suggest to replace "have" with "install".

File: docs/howto/openstack/magnum/new-k8s-cluster.md [hunk 3]
```diff
@@ -0,0 +1,308 @@
...
+
+	```plain
+	+---------+-------------------------------------------------+------+
+	| Name    | Fingerprint                                     | Type |
+	+---------+-------------------------------------------------+------+
+	| husavik | 34:3b:58:ba:ec:95:f5:17:17:df:04:38:11:89:e6:3d | ssh  |
+	+---------+-------------------------------------------------+------+
+	```
+	
+	...and set to the `KEYPAIR` variable the name of the keypair
```
Comment (fghaas) [inline docs/howto/openstack/magnum/new-k8s-cluster.md:115]: Should be:

```markdown
... and set the `KEYPAIR` variable to the name of the keypair
you want:
```

File: docs/howto/openstack/magnum/new-k8s-cluster.md [hunk 4]
```diff
@@ -0,0 +1,308 @@
...
+	you want:
+	
+	```bash
+	KEYPAIR="husavik" # again, this is just an example
+	```
+	
+	Finally, decide on a name for your new Kubernetes cluster:
+	
+	```bash
+	CLUSTER_NAME="bangor" # just remembered a famous storyteller
```
Comment (fghaas) [inline docs/howto/openstack/magnum/new-k8s-cluster.md:125]: The comment can go; maybe there's a Welsh or Irish or American person that remembers Bangor for being their hometown. :)

File: docs/howto/openstack/magnum/new-k8s-cluster.md [hunk 5]
```diff
@@ -0,0 +1,308 @@
...
+		--cluster-template $CLUSTER_TMPL \
+		--keypair husavik
+		bangor
+	```
+	If everything went well with your request for a new cluster, 
+	on your terminal, you would see a message like the following:
+	 
+	 ```plain
+	 Request to create cluster e0df8c62-c6f6-4c7d-b67e-33e3606e9ab6 
+	accepted
```
Comment (fghaas) [inline docs/howto/openstack/magnum/new-k8s-cluster.md:142]: the "accepted" line appears to be oddly indented with one less space?

File: docs/howto/openstack/magnum/new-k8s-cluster.md [hunk 6]
```diff
@@ -0,0 +1,308 @@
...
+	![Cluster details](assets/new-k8s/shot-05.png)
+=== "OpenStack CLI"
+	To list all available Kubernetes clusters, just type:
+	
+	```bash
+	openstack coe cluster list
+	```
+
+	```plain
+	+--------------------------------------+--------+---------+------------+--------------+-----------------+---------------+
```
Comment (fghaas) [inline docs/howto/openstack/magnum/new-k8s-cluster.md:181]: This doesn't look like `CLIFF_MAX_TERM_WIDTH=100` was set.

File: docs/howto/openstack/magnum/new-k8s-cluster.md [hunk 7]
```diff
@@ -0,0 +1,308 @@
...
+
+	For many more details on a specific cluster, note its name and 
+	run a command like this:
+	
+	```bash
+	openstack coe cluster show bangor
+	```
+
+	```plain
+	+----------------------+---------------------------------------------------------------------------------------------------------+
```
Comment (fghaas) [inline docs/howto/openstack/magnum/new-k8s-cluster.md:196]: Here, too, we're in excess of 100 characters.

File: docs/howto/openstack/magnum/new-k8s-cluster.md [hunk 8]
```diff
@@ -0,0 +1,308 @@
...
+	```bash
+	brew install kubectl
+	```
+
+Before running commands against a specific cluster, you must have the 
+corresponding `config` file on your computer.
+
+=== "{{gui}}"
+	Downloading a `config` file from the {{gui}} is currently not 
+	supported. You can still have the `config` file of your newly created
```
Comment (fghaas) [inline docs/howto/openstack/magnum/new-k8s-cluster.md:255]: I would replace "have" with "fetch" here.

File: docs/howto/openstack/magnum/new-k8s-cluster.md [hunk 9]
```diff
@@ -0,0 +1,308 @@
...
+
+	```bash
+	openstack coe cluster config bangor
+	```
+
+After saving the `config` file locally, set the value of variable 
+`KUBECONFIG` to the full path of the file. Type, for example:
+
+```bash
+export KUBECONFIG=${PWD}/config
```
Comment (fghaas) [inline docs/howto/openstack/magnum/new-k8s-cluster.md:270]: If you're using `${PWD}` in the `export` command below, you should also use `openstack coe cluster config --dir=${PWD} bangor`.

**Review by fghaas — COMMENTED** [review]
Overall this looks excellent and very nearly good to merge already. I just have a few nits.

File: docs/howto/openstack/magnum/new-k8s-cluster.md (hunk 1 above)
Comment (colder-is-better) [reply docs/howto/openstack/magnum/new-k8s-cluster.md:15]: Much better, yes!

File: docs/howto/openstack/magnum/new-k8s-cluster.md (hunk 2 above)
Comment (colder-is-better) [reply docs/howto/openstack/magnum/new-k8s-cluster.md:23]: Good suggestion!

File: docs/howto/openstack/magnum/new-k8s-cluster.md (hunk 4 above)
Comment (colder-is-better) [reply docs/howto/openstack/magnum/new-k8s-cluster.md:125]: American :)

File: docs/howto/openstack/magnum/new-k8s-cluster.md (hunk 5 above)
Comment (colder-is-better) [reply docs/howto/openstack/magnum/new-k8s-cluster.md:142]: Yeah, that's due to the `fold` tool and my manual edits :/

File: docs/howto/openstack/magnum/new-k8s-cluster.md (hunk 7 above)
Comment (colder-is-better) [reply docs/howto/openstack/magnum/new-k8s-cluster.md:196]: In my RC. Forever.

File: docs/howto/openstack/magnum/new-k8s-cluster.md (hunk 8 above)
Comment (colder-is-better) [reply docs/howto/openstack/magnum/new-k8s-cluster.md:255]: Much better, yes.

File: docs/howto/openstack/magnum/new-k8s-cluster.md (hunk 9 above)
Comment (colder-is-better) [reply docs/howto/openstack/magnum/new-k8s-cluster.md:270]: Done!

**Comment by fghaas** [comment]
As far as these additions go, they look excellent and I have nothing to add or change.

I have two more suggestions, though:

* Add a `.pages` file in the `howto/magnum` directory, for which I would suggest the following content:
   ```yaml
   title: "Magnum (Kubernetes cluster management)"
   nav:
     - new-k8s-cluster.md
   ```
* Create a link to this howto from `howto/kubernetes/index.md`, specifically from its "Magnum" subsection. This might require a slight rewrite of those two paragraphs.

**Comment by colder-is-better** [comment]
Aha, understood. I have already added `docs/howto/openstack/magnum/.pages` and populated the new file accordingly. Not sure how to handle `docs/howto/kubernetes/index.md`, though; should I add the new (modified) version of it in this PR, or should I create a new one?

**Comment by fghaas** [comment]
> should I add the new (modified) version of it in this PR, or should I create a new one?

You can include it in this PR.

File: docs/howto/kubernetes/index.md [hunk 10]
```diff
@@ -1,30 +1,41 @@
...
-in front etc.
+Magnum lets you create clusters via the OpenStack APIs. To do that, you 
+base your configuration on a Cluster Template. In the template
+you define parameters that describe how the cluster will be 
+constructed, for example worker flavors. In each our regions we offer
+predefined public Cluster Templates with ready to use configurations. 
+To learn more about the templates, check out the
+[Magnum documentation](https://docs.openstack.org/magnum/latest/user/#clustertem
+plate).
+
```
Comment (fghaas) [inline docs/howto/kubernetes/index.md:36]: I don't think we want to leave this bit in. That's what I meant by saying that this needed a bit of a rewrite. Let's remove any references to _creating_ a template, and just include the reference to the how-to that talks about creating a cluster _using_ an existing template.

Comment (colder-is-better) [reply docs/howto/kubernetes/index.md]: Aha -- got it and updated accordingly!

**Review by fghaas — APPROVED** [review]
This looks good to me now! I'll rebase on the current state of `main`, and then merge this.

**Review by fghaas — DISMISSED** [review]

**Comment by fghaas** [comment]
Something is a bit fishy with the GitHub Actions infrastructure today, it seems. Evidently the CI runs have completed without error but they are not reporting back. I'll try rerunning the jobs.

**Pull request merged**